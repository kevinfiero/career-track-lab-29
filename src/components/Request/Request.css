.Request {
  margin: 2rem;
  height: 10rem;
  width: 40rem;
  border: 5px solid black;
  display: flex;
}

.Request .input{
  display: block;
  align-self: center;
  position: absolute;
  width: 4rem;
  height: 1rem;
  z-index: 100;
  cursor: pointer;
  opacity: 0.01;
}

.Request .input[type="radio"]:checked+label{
  color: blue;
  z-index: 100;
  cursor: pointer;
}

.Request .label{
  z-index: 1;
  width: 4rem;

}

.Request .button{
  width: 4rem;
  height: 1.5rem;
  z-index: 1;
  border: 1px solid #CCC;
  text-align: center;
  margin: 1rem .5rem 1rem .5rem;
}
