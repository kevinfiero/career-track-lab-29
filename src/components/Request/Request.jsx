import React from 'react';
import PropTypes from 'prop-types';
import styles from './Request.css';

const Request = ({ button, url, inputField, onChange, onClick }) => (
  <div className = {styles.Request}>
    <div className = {styles.row}>
      <div className={styles.center}>
        <div className={styles.radiobutton}>
          <input 
            type="radio" 
            name="button"
            value="GET"
            checked={button === 'GET'}
            onChange={onChange}
            className={styles.input}
          />
          <label className={styles.label}>GET</label>
        </div>
        <div className={styles.radiobutton}>
          <input 
            type="radio" 
            name="button"
            value="POST"  
            checked={button === 'POST'}
            onChange={onChange}
            className={styles.input}
          />
          <label className={styles.label}>POST</label>
        </div>
        <div className={styles.radiobutton}>
          <input 
            type="radio" 
            name="button"
            value="PUT"  
            checked={button === 'PUT'}
            onChange={onChange}
            className={styles.input}
          />
          <label className={styles.label}>PUT</label>
        </div>
        <div className={styles.radiobutton}>
          <input 
            type="radio" 
            name="button"
            value="DELETE"
            checked={button === 'DELETE'}
            onChange={onChange}
            className={styles.input}
          />
          <label className={styles.label}>DELETE</label>
        </div>
      </div>
      <div className={styles.column}>
        <div className={styles.row}>
          <input 
            type="text" 
            name="url" 
            value={url} 
            onChange={onChange} 
            className={styles.urlinput}
          />
          <button type="button" 
            onClick={onClick} 
            className={styles.submitbutton}>
            Submit
          </button>
        </div>
        <textarea 
          name="inputField" 
          value={inputField} 
          onChange={onChange} 
          className={styles.textinput}/>
      </div>
    </div>
  </div>
);

Request.propTypes = {
  button: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  inputField: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired
};

export default Request;
