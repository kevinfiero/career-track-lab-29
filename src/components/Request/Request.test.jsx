import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Request from './Request';

describe('Request component', () => {
  afterEach(() => cleanup());
  it('renders Request', () => {
    const { asFragment } = render(<Request
      button="POST"
      url="https://reqres.in/api/users"
      inputField= '{"name": "sasa","job": "leader"}'
      onChange={jest.fn()}
      onClick={jest.fn()}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
