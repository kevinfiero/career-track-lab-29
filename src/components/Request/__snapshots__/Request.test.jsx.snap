// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request component renders Request 1`] = `
<DocumentFragment>
  <div
    class="Request"
  >
    <div
      class="row"
    >
      <div
        class="center"
      >
        <div
          class="radiobutton"
        >
          <input
            class="input"
            name="button"
            type="radio"
            value="GET"
          />
          <label
            class="label"
          >
            GET
          </label>
        </div>
        <div
          class="radiobutton"
        >
          <input
            checked=""
            class="input"
            name="button"
            type="radio"
            value="POST"
          />
          <label
            class="label"
          >
            POST
          </label>
        </div>
        <div
          class="radiobutton"
        >
          <input
            class="input"
            name="button"
            type="radio"
            value="PUT"
          />
          <label
            class="label"
          >
            PUT
          </label>
        </div>
        <div
          class="radiobutton"
        >
          <input
            class="input"
            name="button"
            type="radio"
            value="DELETE"
          />
          <label
            class="label"
          >
            DELETE
          </label>
        </div>
      </div>
      <div
        class="column"
      >
        <div
          class="row"
        >
          <input
            class="urlinput"
            name="url"
            type="text"
            value="https://reqres.in/api/users"
          />
          <button
            class="submitbutton"
            type="button"
          >
            Submit
          </button>
        </div>
        <textarea
          class="textinput"
          name="inputField"
        >
          {"name": "sasa","job": "leader"}
        </textarea>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
